src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!this.isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
That's ok, but I don't understand why iteration over all tasks is necessary in this context. Because if the current task is part of the task pool, it must be non-idle element, or